arch/xtensa/src/esp32s3/esp32s3_sdmmc.c

```diff
@@ -0,0 +1,2924 @@
+  worker_t           callback;        /* Registered callback function */
+  void              *cbarg;           /* Registered callback argument */
+  struct work_s      cbwork;          /* Callback work queue structure */
+
+  /* Interrupt mode data transfer support */
+
+  uint32_t          *buffer;          /* Address of current R/W buffer */
+  size_t             remaining;       /* Number of bytes remaining in the transfer */
+  uint32_t           xfrmask;         /* Interrupt enables for data transfer */
+#ifdef CONFIG_ESP32S3_SDMMC_DMA
+  uint32_t           dmamask;         /* Interrupt enables for DMA transfer */
+  volatile struct sdmmc_dma_s dma_desc[NUM_DMA_DESCRIPTORS];
+#ifdef CONFIG_ESP32S3_SPIRAM
+  uint8_t           *dma_buf;
+  size_t            dma_buf_size;
+#endif
+#endif
+  bool               wrdir;           /* True: Writing False: Reading */
+
+  /* DMA data transfer support */
+
+  int                slot;
+
+  const sdmmc_slot_io_info_t *sdio_pins;
+  const sdmmc_slot_info_t *slot_info;
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/* #define CONFIG_ESP32S3_SDMMC_REGDEBUG */
+
+#ifdef CONFIG_ESP32S3_SDMMC_REGDEBUG
+static uint32_t __esp32s3_getreg(const char *func, uint32_t addr);
+static void __esp32s3_putreg(const char *func, uint32_t val, uint32_t addr);
+
+#  define esp32s3_getreg(addr)     __esp32s3_getreg(__func__, addr)
+#  define esp32s3_putreg(val,addr) __esp32s3_putreg(__func__, val,addr)
+#else
+#  define esp32s3_getreg(addr)     getreg32(addr)
+#  define esp32s3_putreg(val,addr) putreg32(val,addr)
+#endif
+
+/* Low-level helpers ********************************************************/
+
+/* DMA Helpers **************************************************************/
+
+/* Data Transfer Helpers ****************************************************/
+
+static void esp32s3_eventtimeout(wdparm_t arg);
+static void esp32s3_endwait(struct esp32s3_dev_s *priv,
+                          sdio_eventset_t wkupevent);
+static void esp32s3_endtransfer(struct esp32s3_dev_s *priv,
+                              sdio_eventset_t wkupevent);
+
+/* Interrupt Handling *******************************************************/
+
+static int  esp32s3_interrupt(int irq, void *context, void *arg);
+
+/* SDIO interface methods ***************************************************/
+
+/* Mutual exclusion */
+
+#ifdef CONFIG_SDIO_MUXBUS
+static int esp32s3_lock(struct sdio_dev_s *dev, bool lock);
+#endif
+
+/* Initialization/setup */
+
+static void esp32s3_reset(struct sdio_dev_s *dev);
+static sdio_capset_t esp32s3_capabilities(struct sdio_dev_s *dev);
+static sdio_statset_t esp32s3_status(struct sdio_dev_s *dev);
+static void esp32s3_widebus(struct sdio_dev_s *dev, bool enable);
+static void esp32s3_clock(struct sdio_dev_s *dev,
+              enum sdio_clock_e rate);
```

Comment:
If possible, keep arguments aligned in different lines. Please check other occurrences in this file.

```suggestion
static void esp32s3_clock(struct sdio_dev_s *dev,
                                              enum sdio_clock_e rate);
```